streamrouter/router.py

```diff
@@ -1,34 +1,105 @@
-import hmac_tokens
-import mmh3
+import asyncio
 import re
-import time
-import logging
-import os
 
-from streamrouter import ConsulClient
+from contextlib import contextmanager
 
-logger = logging.getLogger(__name__)
+from .config import RouterConfig
+from .consul import (
+    ArrowAsnsService, Consul, HlsEdgeService, MjpegProxyService, Mp4EdgeService,
+    RtspconService, SynchronizationError,
+)
+from .native import NativeObject, get_stream_router_lib, get_string
 
+lib = get_stream_router_lib()
 
-class NotImplementedStream(Exception):
-    def __init__(self, request_format, possible_formats):
-        self.request_stream_format = request_format
-        self.possible_formats = possible_formats
 
+@contextmanager
+def native_router_config_factory(config):
+    """
+    A context manager that yields native router configuration object
+    initialized from a given RouterConfig.
+    """
+    assert type(config) is RouterConfig
```

Comment:
`assert isinstance(config, RouterConfig)` is more pythonic - same also on the next line

Reply:
I'm not sure if it's more pythonic. It's simply a different thing. Here I'm checking if the `config` object is exactly of the `RouterConfig` class (i.e. not a subclass). The `isinstance` construction would also accept any subclass. Anyway, both constructions would work here. I was just too lazy to type it down and I copied it from a different part of code. :D